core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -127,16 +127,24 @@ class PlaintextConsumerTest extends BaseConsumerTest {
 
   @Test
   def testHeadersExtendedSerializerDeserializer(): Unit = {
-    val extendedSerializer = new ExtendedSerializer[Array[Byte]] with SerializerImpl
-    val extendedDeserializer = new ExtendedDeserializer[Array[Byte]] with DeserializerImpl
+    val extendedSerializer = new ExtendedSerializer[Array[Byte]] with SerializerImpl {
+      override def serialize(topic: String, headers: Headers, data: Array[Byte]): Array[Byte] = super.serialize(topic, headers, data)
```

Comment:
I wonder if it would be clearer to rename the `serialize` method in `SerializerImpl` so there is no confusion about which parent method is being invoked?